src/app/Components/shopping-list/shopping-list.component.ts

```diff
@@ -59,23 +70,29 @@ export class ShoppingListComponent implements OnInit {
     if (this.shoppingForm.invalid) {
       return;
     }
-
     const ingredient = this.shoppingForm.value;
-
     this.dataShopList.updateIngredient(ingredient);
     this.shoppingForm.reset({ name: '', amount: 0 });
     this.modeUse = false;
   }
 
   deleteIngredient() {
     const id = this.shoppingForm.value.id;
+    console.log(id);
 
     this.dataShopList.deleteIngredient(id);
+    this.dataShopList.ingredients.pipe(take(1)).subscribe((res) => {
```

Comment:
no need cuz in ngOnInit, the stream of `this.dataShopList.ingredients` has been subscribed, which means for every new value change, it will assign that value to `this.ingredients`